### PR TITLE
refactor(audit): move audit report rendering from cmd to converter/builder layer

### PR DESCRIPTION
**Pull request by unclesp1d3r**
## Summary

Moves compliance audit report rendering from `cmd/audit_handler.go` into the `internal/converter/builder/` layer, replaces the stub `ComplianceChecks` model with a rich `ComplianceResults` structure, simplifies Mergify merge queues, and adds comprehensive doc comments across the codebase.

**Impact**: 54 files changed (+1,815 / -798)

| Area | Files | +/- | Description |
|------|-------|-----|-------------|
| Audit rendering | 13 | +1,574 / -742 | Core refactor (issue #322) |
| Mergify | 1 | +105 / -37 | Queue simplification |
| Docstrings | 37 | +91 / -11 | Doc comment coverage |
| Docs | 3 | +45 / -8 | AGENTS.md + model reference |

### 1. Audit Report Rendering (issue #322)

Moved inline markdown generation out of the CLI handler into the converter/builder layer, making audit reports work across all output formats (markdown, JSON, YAML) without format-specific code in the handler.

| Area | Change |
|------|--------|
| `cmd/audit_handler.go` | Removed `appendAuditFindings` / `appendPluginFindings`. Now maps `audit.Report` to `device.ComplianceChecks` via `mapAuditReportToComplianceResults()` and delegates to `generateWithProgrammaticGenerator()`. Creates a shallow copy of `*CommonDevice` before setting `ComplianceChecks` (immutability). |
| `cmd/audit_handler_test.go` | Rewrote tests to validate mapping logic instead of markdown string output |
| `internal/converter/builder/builder.go` | Added `BuildAuditSection()` -- renders compliance summary, plugin results, findings tables, and metadata |
| `internal/converter/builder/helpers.go` | Added `EscapePipeForMarkdown()` and `TruncateString()` (rune-aware) helpers |
| `internal/converter/builder/writer.go` | Added `WriteAuditSection()` streaming variant |
| `internal/converter/builder/*_test.go` | Full test coverage for builder, helpers, and writer |
| `internal/converter/hybrid_generator.go` | Wired `BuildAuditSection()` / `WriteAuditSection()` into markdown generation when `ComplianceChecks != nil` |
| `internal/converter/enrichment.go` | Pass-through for `ComplianceChecks` in `prepareForExport()` |
| `internal/model/common/enrichment.go` | Replaced stub `ComplianceChecks` with rich `ComplianceResults`, `ComplianceFinding`, `PluginComplianceResult`, `ComplianceControl`, `ComplianceResultSummary`, `CompliancePluginInfo`, `ComplianceAttackSurface` |
| `internal/model/common/enrichment_test.go` | Table-driven tests for `ComplianceResults.HasData()` |
| `internal/model/common/device.go` | Changed `ComplianceChecks` field type from `*ComplianceChecks` to `*ComplianceResults` |

**Key design decisions:**
- `common.ComplianceResults` mirrors `audit.Report` shape but lives in `common` package (avoids circular deps with `audit`)
- `handleAuditMode()` no longer contains format-specific code -- JSON/YAML serialize `ComplianceChecks` automatically via struct tags
- `BuildAuditSection()` / `WriteAuditSection()` return empty string / nil when data is nil -- safe to call unconditionally
- `BuildAuditSection` has a fallback total findings calculation when `Summary` is nil (sums direct + per-plugin findings)
- All map iterations use `slices.Sorted(maps.Keys(...))` for deterministic output
- `ComplianceResultSummary` int fields use `json:"field"` (no `omitempty`) to distinguish zero from absent

### 2. Mergify Queue Simplification

Replaced 5 author-specific queues (dosubot, dependabot-workflows, dependabot, maintainer, external) with 4 queues -- consolidating maintainer + external into a single manually-enqueued `default` queue gated by repo permissions.

| Queue | Author | Entry | CI Required |
|-------|--------|-------|-------------|
| `dosubot` | `dosubot[bot]` | autoqueue | Lint only |
| `dependabot-workflows` | `dependabot[bot]` (workflow-only files) | autoqueue | Lint only |
| `dependabot` | `dependabot[bot]` (catchall) | autoqueue | Full CI |
| `default` | everyone else | manual `/queue` command | Full CI |

- No `lgtm` label, `do-not-merge` label, or approval conditions on human queues
- Repo permissions gate who can send `/queue` -- simpler than label-based access control
- `merge_protections` enforce CI independently -- even manually enqueued PRs must pass all checks
- Conventional commit regex updated to allow breaking-change `!` marker (e.g., `feat(api)!: redesign`)

### 3. Doc Comment Coverage

Added ~67 doc comments across 37 files to improve godoc coverage. All previously undocumented `func`, `type`, `var`, `const`, and const blocks now have doc comments following Go convention (start with the declaration name).

Key packages covered: `cmd/`, `internal/diff/`, `internal/converter/`, `internal/converter/builder/`, `internal/converter/formatters/`, `internal/plugins/sans/`, `internal/plugins/stig/`, `internal/processor/`, `internal/validator/`, `internal/sanitizer/`, `internal/markdown/`, `internal/analysis/`, `internal/audit/`, `internal/model/`, `internal/schema/`, `internal/cfgparser/`, `internal/display/`, `internal/export/`.

### 4. Documentation Updates

- **AGENTS.md**: Updated repository roles, Mergify queue docs, `BuildAuditSection` behavior, `ComplianceResult` nested map documentation
- **docs/data-model/model-reference.md**: Updated for `ComplianceChecks` -> `ComplianceResults` rename
- **docs/dev-guide/api.md**: Fixed `ComplianceResult` summary field formatting

### Audit rendering
- [x] `just lint` -- 0 issues
- [x] `just test` -- all pass
- [x] Builder unit tests: nil `ComplianceChecks`, empty findings, populated findings with pipe escaping, metadata rendering
- [x] Writer streaming tests: nil data, populated data
- [x] Helper tests: pipe escaping edge cases, truncation with multi-byte chars (rune-aware), max description length
- [x] `handleAuditMode` mapping tests: field-by-field correctness
- [x] `ComplianceResults.HasData()` tests: zero value, mode, findings, plugin results, summary, metadata

### Mergify
- [ ] Verify config validates (Mergify dashboard or next bot PR)
- [ ] Confirm dosubot PRs auto-approve and merge with lint-only
- [ ] Confirm dependabot workflow PRs use `dependabot-workflows` queue (lint-only)
- [ ] Confirm dependabot non-workflow PRs use `dependabot` queue (full CI)
- [ ] Confirm human PRs require manual `/queue` + full CI via merge protections

### Docstrings
- [x] Build passes (`go build ./...`)
- [x] No lint regressions
- [x] No test regressions

---

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -1,6 +1,83 @@
...
+  # 3. dependabot — non-workflow changes, full CI
+  # ─────────────────────────────────────────────────────────────────────────
+  - name: dependabot
+    merge_method: squash
+    autoqueue: true
+    queue_conditions:
+      - base = main
+      - label != do-not-merge
+      - author = dependabot[bot]
+    merge_conditions:
```
[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply .mergify.yml]: Not applicable -- Mergify evaluates `queue_rules` in definition order and assigns the PR to the first matching queue. The `dependabot-workflows` queue (lint-only) is defined before `dependabot` (full CI) and has a stricter `files` condition that catches workflow-only PRs first. The `dependabot` queue acts as a fallback for all other dependabot PRs, so there's no overlap risk without an additional files exclusion.
